File: java/rooms/RoomFactory.java
package rooms;

import chest.ChestType;
import monsters.CreatureType;
import rooms.options.GameMasterable;
import rooms.options.RoomType;

public interface RoomFactory {
    GameMasterable createEmptyRoom(RoomType roomType);
    GameMasterable createRoomWithChest(RoomType roomType, ChestType chestType);
    GameMasterable createRoomWithCreature(RoomType roomType, CreatureType creatureType);
}

File: php/rooms/RoomFactory.php
<?php
namespace rooms;

use chest\ChestType;
use monsters\CreatureType;
use rooms\options\GameMasterable;
use rooms\options\RoomType;

interface RoomFactory {
    public function createEmptyRoom(RoomType $roomType): GameMasterable;
    public function createRoomWithChest(RoomType $roomType, ChestType $chestType): GameMasterable;
    public function createRoomWithCreature(RoomType $roomType, CreatureType $creatureType): GameMasterable;
}
?>